test/distributed/org/apache/cassandra/distributed/test/sai/AnalzyerDistributedTest.java

```diff
@@ -0,0 +1,140 @@
+{
+    @Rule
+    public SAITester.FailureWatcher failureRule = new SAITester.FailureWatcher();
+
+    private static final String CREATE_KEYSPACE = "CREATE KEYSPACE %%s WITH replication = {'class': 'SimpleStrategy', 'replication_factor': %d}";
+    private static final String CREATE_TABLE = "CREATE TABLE %%s (pk int PRIMARY KEY, not_analyzed int, val text)";
+    private static final String CREATE_INDEX = "CREATE CUSTOM INDEX ON %%s(%s) USING 'StorageAttachedIndex' WITH OPTIONS = {'index_analyzer': 'standard'}";
+    private static final int NUM_REPLICAS = 3;
+    private static final int RF = 2;
+
+    private static final AtomicInteger seq = new AtomicInteger();
+    private static String table;
+
+    private static Cluster cluster;
+
+    private static int dimensionCount;
+
+    @BeforeClass
+    public static void setupCluster() throws Exception
+    {
+        cluster = Cluster.build(NUM_REPLICAS)
+                         .withConfig(config -> config.with(GOSSIP).with(NETWORK))
+                         .start();
+
+        cluster.schemaChange(withKeyspace(String.format(CREATE_KEYSPACE, RF)));
+    }
+
+    @AfterClass
+    public static void closeCluster()
+    {
+        if (cluster != null)
+            cluster.close();
+    }
+
+    @Before
+    public void before()
+    {
+        table = "table_" + seq.getAndIncrement();
+        dimensionCount = getRandom().nextIntBetween(100, 2048);
+    }
+
+    @After
+    public void after()
+    {
+        cluster.schemaChange(formatQuery("DROP TABLE IF EXISTS %s"));
+    }
+
+    @Test
+    public void testAnalyzerSearch()
+    {
+        cluster.schemaChange(formatQuery(String.format(CREATE_TABLE, dimensionCount)));
+        cluster.schemaChange(formatQuery(String.format(CREATE_INDEX, "val")));
+        SAIUtil.waitForIndexQueryable(cluster, KEYSPACE);
+
+        var iterations = 15000;
+        for (int i = 0; i < iterations; i++)
+        {
+            var x = i % 100;
+            if (i % 100 == 0)
+            {
+                execute(String.format(
+                "INSERT INTO %s (pk, not_analyzed, val) VALUES (%s, %s, '%s')",
+                KEYSPACE + '.' + table, i, x, "this will be tokenized"));
+            }
+            else if (i % 2 == 0)
+            {
+                execute(String.format(
+                "INSERT INTO %s (pk, not_analyzed, val) VALUES (%s, %s, '%s')",
+                KEYSPACE + '.' + table, i, x, "this is different"));
+            }
+            else
+            {
+                execute(String.format(
+                "INSERT INTO %s (pk, not_analyzed, val) VALUES (%s, %s, '%s')",
+                KEYSPACE + '.' + table, i, x, "basic test"));
+            }
+        }
+        var result = execute("SELECT * FROM %s WHERE val : 'tokenized'");
+        assertThat(result).hasSize(iterations / 100);
+        result = execute("SELECT * FROM %s WHERE val : 'this'");
+        assertThat(result).hasSize(iterations / 2);
```

Comment:
Maybe add a comment about what you are expecting to match?  Like this is going to match both "this will be tokenized" and "this is different".  Took me a double take to get that right in my head and the expected numbers ...

Reply:
Do you mind if I fix this in a subsequent PR so that I can get this built asap?

Reply:
WFM

Reply:
PR for adding comments: https://github.com/datastax/cassandra/pull/733